rear_rider_device/bluetooth_server_child_proc.py

```diff
@@ -93,21 +92,21 @@ async def on_discoverable(self):
         """
         The handler for when bluetooth changes its discoverability to other devices.
         """
-        line = await self.readline()
-        if line == '1':
-            await self._leds_child_process.set_discoverable_effect(False)
-        elif line == '0':
+        vals = (await self.readline()).split(' ')
+        discoverable = vals[0]
+        self._print(f'Discoverable: {discoverable} ; Timeout: {vals[1]}')
+        if discoverable == '1':
+            await self._leds_child_process.set_discoverable_effect(True)
+        elif discoverable == '0':
             await self._leds_child_process.set_discoverable_effect(False)
```

Comment:
non-blocking: 

is discoverable ever a value besides 1 or 0? if not, you can just pass `discoverable` and omit the whole `if` block:

```suggestion
            await self._leds_child_process.set_discoverable_effect(discoverable)
```

a python bool is a subset of integers, meaning 1 == True and 0 == False

Reply:
I know technically this could work, but I usually do not do that since having False or True explicitly states the intention of the function call without having to pass around the variable. Also the discoverable variable is a string, which is also why I did what I did.